common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts

```diff
@@ -393,15 +393,12 @@ export class LDMLKeyboardXMLSourceFileReader {
         r.stuffBoxes(test, $$, 'startContext'); // singleton
         // now the actions
         test.actions = $$.map(v => {
-          const subtag = v['#name'];
-          const subv = LDMLKeyboardXMLSourceFileReader.defaultMapper(v, r);
-          switch(subtag) {
-            case 'keystroke': return { keystroke: subv };
-            case 'check':     return { check:     subv };
-            case 'emit':      return { emit:      subv };
-            case 'startContext': return null; // handled above
-            default: this.callbacks.reportMessage(CommonTypesMessages.Error_TestDataUnexpectedAction({ subtag })); return null;
+          const type = v['#name']; // element name
+          if (type === 'startContext') {
+            return null; // handled above
           }
+          const subv = LDMLKeyboardXMLSourceFileReader.defaultMapper(v, r);
+          return Object.assign({ type }, subv);
```

Comment:
This will return an object similar to `{keystroke: subv}` ? Seems like it should be different. , `{ type: "keystroke", (other props from subv) }`

Reply:
The old format was one level deeper in objects, wasn't it?

Reply:
yes. does the test fail locally?

Reply:
I was running the wrong test.